src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -130,9 +167,46 @@ export default function withAnnotations<P extends object>(
             }
         };
 
+        handleAnnotationsFetchError = ({ error }: { error: ElementsError }): void => {
+            if (!error) {
+                return;
+            }
+
+            this.setState({
+                error: {
+                    ...error,
+                    code: ERROR_CODE_FETCH_ANNOTATIONS,
+                },
+            });
+
+            this.escalateAnnotationsFetchError(error);
+        };
+
         handleAnnotator = (annotator: Annotator): void => {
+            const { error } = this.state;
+
             this.annotator = annotator;
             this.annotator.addListener('annotatorevent', this.handleAnnotatorEvent);
+
+            if (error && 'origin' in error && error.origin === ORIGIN_ACTIVITY_SIDEBAR) {
+                this.handleSidebarFetchError(error);
+            }
+        };
+
+        handleAnnotatorCreate = (annotator: Annotator): void => {
+            annotator.addListener('annotations_fetch_error', this.handleAnnotationsFetchError);
+        };
+
+        handleSidebarFetchError = (error: ElementsError): void => {
+            const { annotator } = this;
+
+            if (!annotator) {
+                this.setState({ error });
+                return;
+            }
+
+            this.escalateAnnotationsFetchError(error);
+            annotator.emit('annotations_visible_set', false);
```

Comment:
How would this handle the scenario where the sidebar fetch failed before the annotator was emitted? Would it make it easier to reason about if we added some state for `hasAnnotator` or something similar and in `componentDidUpdate`:

- `if (hasAnnotator && error)` then 
- escalate the error
- `annotator.emit('annotations_visible_set', false)`

Though that would potentially require separating the component state from `AnnotatorState`

Reply:
If sidebar fetch failed before annotator was emitted, `handleSidebarFetchError` would set error in state. And when annotator is emitted, `handleAnnotator` will check the error in state and call `handleSidebarFetchError` again.

Reply:
Yeah, that'll simplify the logic. Updated.